nano/node/bootstrap/message_deserializer.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <nano/node/common.hpp>
+#include <nano/node/socket.hpp>
+
+#include <boost/system/error_code.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace nano
+{
+class socket;
+namespace bootstrap
+{
+	class message_deserializer : public std::enable_shared_from_this<nano::bootstrap::message_deserializer>
+	{
+	public:
+		enum class parse_status
+		{
+			none,
+			success,
+			insufficient_work,
+			invalid_header,
+			invalid_message_type,
+			invalid_keepalive_message,
+			invalid_publish_message,
+			invalid_confirm_req_message,
+			invalid_confirm_ack_message,
+			invalid_node_id_handshake_message,
+			invalid_telemetry_req_message,
+			invalid_telemetry_ack_message,
+			invalid_bulk_pull_message,
+			invalid_bulk_pull_account_message,
+			invalid_frontier_req_message,
+			invalid_network,
+			outdated_version,
+			duplicate_publish_message,
+			message_size_too_big,
+		};
+
+		using callback_type = std::function<void (boost::system::error_code, std::unique_ptr<nano::message>)>;
```

Comment:
I think passing `parse_status` is better than passing boost::system::error_code.
The boost library should be an implementation detail at this level.

Reply:
We can do this on another PR since  there would be a lot of knock on effects